packages/react-reconciler/src/ReactFiberTreeReflection.js

```diff
@@ -333,13 +333,21 @@ export function findCurrentHostFiberWithNoPortals(parent: Fiber): Fiber | null {
   return null;
 }
 
+// This function detects when a Suspense boundary goes from visible to hidden.
+// It returns false if the boundary is already hidden.
+// TODO: Use an effect tag. And move this to ReactFiberCommitWork.
 export function isFiberSuspenseAndTimedOut(fiber: Fiber): boolean {
```

Comment:
This function is also used by Scopes for traversal, will this change affect that too? See:

https://github.com/facebook/react/blob/master/packages/react-reconciler/src/ReactFiberScope.new.js#L51

Reply:
My mistake, didn't search for other uses.

I moved the logic into the commit phase.